geode-core/src/main/java/org/apache/geode/cache/query/security/JavaBeanAccessorMethodAuthorizer.java

```diff
@@ -0,0 +1,153 @@
+   * (see {@link Declarable#initialize(Cache, Properties)}), when using a declarative approach.
+   *
+   * @param cache the {@code Cache} instance that owns this authorizer, required in order to
+   *        configure the default {@link RestrictedMethodAuthorizer}.
+   * @param allowedPackages the packages containing classes for which 'is' and 'get' methods will
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(Cache cache, Set<String> allowedPackages) {
+    Objects.requireNonNull(cache, NULL_CACHE_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = new RestrictedMethodAuthorizer(cache);
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Creates a {@code JavaBeanAccessorMethodAuthorizer} object and initializes it so it can be
+   * safely used in a multi-threaded environment.
+   * <p/>
+   *
+   * @param restrictedMethodAuthorizer the default {@code RestrictedMethodAuthorizer} to use.
+   * @param allowedPackages the packages containing classes for which 'is' and 'get' methods will
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(RestrictedMethodAuthorizer restrictedMethodAuthorizer,
+      Set<String> allowedPackages) {
+    Objects.requireNonNull(restrictedMethodAuthorizer, NULL_AUTHORIZER_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = restrictedMethodAuthorizer;
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Executes the authorization logic to determine whether the {@code method} is allowed to be
+   * executed on the {@code target} object instance.
+   * <p/>
+   *
+   * @param method the {@link Method} that should be authorized.
+   * @param target the {@link Object} on which the {@link Method} will be executed.
+   * @return {@code true} if the {@code method} can be executed on on the {@code target} instance,
+   *         {@code false} otherwise.
+   *
+   * @see org.apache.geode.cache.query.security.MethodInvocationAuthorizer
+   */
+  @Override
+  public boolean authorize(Method method, Object target) {
+    String methodName = method.getName().toLowerCase();
+    String packageName = target.getClass().getPackage().getName().toLowerCase();
+
+    // Return false for known dangerous methods.
+    if (restrictedMethodAuthorizer.isKnownDangerousMethod(method, target)) {
+      return false;
+    }
+
+    boolean matches = false;
+
+    if ((methodName.startsWith("get") || methodName.startsWith("is"))) {
+      Iterator<String> iterator = this.allowedPackages.iterator();
+      while (iterator.hasNext() && !matches) {
+        matches = iterator.next().startsWith(packageName);
+      }
+    }
+    // Return true if we found a match, otherwise, delegate to the default authorizer.
+    return matches || restrictedMethodAuthorizer.authorize(method, target);
```

Comment:
I've just thought about this scenario while implementing the `RegExMethodAuthorizer`...
What if the user sets `org.apache.geode` as one of the allowed packages for `JavaBeanAccessorMethodAuthorizer`, `matches` will be `true` at this point **BUT** the actual method is not considered safe by Geode (`restrictedMethodAuthorizer.isAllowedGeodeMethod(method, target) = false`)?. With the current implementation the actual return value would be `true`, but the actual result should be `false`... we should either reject any package that starts with `org.apache.geode` during the initialization of the authorizer, or make sure we call `restrictedMethodAuthorizer.isAllowedGeodeMethod(method, target)` when needed.

Reply:
Added changes to address this in the latest commit